modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -15,13 +15,309 @@
 
 import os
 
+import io
+import fsspec
+from fsspec.core import split_protocol, url_to_fs
+from fsspec.registry import get_filesystem_class
+import numpy as np
+from pandas.io.common import is_fsspec_url
+from packaging import version
+
+from modin.core.storage_formats.pandas.utils import compute_chunksize
+from modin.config import NPartitions
+
+
 from modin.core.io.column_stores.column_store_dispatcher import ColumnStoreDispatcher
 from modin.utils import import_optional_dependency
 
 
 class ParquetDispatcher(ColumnStoreDispatcher):
     """Class handles utils for reading `.parquet` files."""
 
+    @classmethod
+    def get_fsspec_files(cls, path, storage_options):
+        """
+        Retrieve filesystem interface and list of files from path.
+
+        Parameters
+        ----------
+        path : str, path object or file-like object
+            Path to dataset.
+        storage_options : dict
+            Parameters for specific storage engine.
+
+        Returns
+        -------
+        filesystem: Any
+            Protocol implementation of registry.
+        files: list
+            List of files from path.
+        """
+        # Older versions of fsspec doesn't support unstrip_protocol(). It
+        # was only added relatively recently:
+        # https://github.com/fsspec/filesystem_spec/pull/828
+        def _unstrip_protocol(protocol, path):
+            protos = (protocol,) if isinstance(protocol, str) else protocol
+            for protocol in protos:
+                if path.startswith(f"{protocol}://"):
+                    return path
+            return f"{protos[0]}://{path}"
+
+        if isinstance(path, io.IOBase):
```

Comment:
Not all `IOBase` instances have an `fs`. Should we be checking for the fsspec IOBase subclass instead?

Reply:
Yeah we should be checking for `AbstractBufferedFile`